File: Kotlin/app/src/main/java/com/example/tictactoe/ui/theme/Theme.kt
package com.example.tictactoe.ui.theme

import androidx.compose.foundation.isSystemInDarkTheme
import androidx.compose.runtime.Composable
import androidx.compose.material3.*

private val LightColors = lightColorScheme(
    primary = Yellow200,
    onPrimary = TextColor,
    background = BackgroundColorLightMode,
    onBackground = TextColorLightMode
)

private val DarkColors = darkColorScheme(
    primary = Yellow200,
    onPrimary = TextColorLightMode,
    background = BackgroundColorDarkMode,
    onBackground = TextColorDarkMode
)

@Composable
fun TicTacToeTheme(
    darkTheme: Boolean = isSystemInDarkTheme(),
    content: @Composable () -> Unit
) {
    val colors = if (darkTheme) DarkColors else LightColors
    MaterialTheme(
        colorScheme = colors,
        typography = Typography,
        content = content
    )
}

File: flutter_tictactoe/lib/ui/theme/theme.dart
import 'package:flutter/material.dart';
import 'colors.dart';
import 'typography.dart';

class AppTheme {
  static ThemeData lightTheme = ThemeData(
    colorScheme: ColorScheme.light(
      primary: AppColors.yellow200,
      onPrimary: AppColors.textColorLightMode,
      background: AppColors.backgroundColorLightMode,
      onBackground: AppColors.textColorLightMode,
    ),
    textTheme: AppTypography.textTheme,
  );

  static ThemeData darkTheme = ThemeData(
    colorScheme: ColorScheme.dark(
      primary: AppColors.yellow200,
      onPrimary: AppColors.textColorLightMode,
      background: AppColors.backgroundColorDarkMode,
      onBackground: AppColors.textColorDarkMode,
    ),
    textTheme: AppTypography.textTheme,
  );

  static ThemeData getTheme(bool isDarkMode) {
    return isDarkMode ? darkTheme : lightTheme;
  }
}
